src/components/FeedbackForm.vue

```diff
@@ -0,0 +1,240 @@
+            type="checkbox" 
+            name="agreement" 
+            class="checkbox-input"
+            @change="formDataValidFlags.agreement = formData.agreement"
+          />
+          <label for="formAgree" class="form-label">
+            I agree with using my personal data
+          </label>
+        </div>
+      </div>
+      <div class="feedback-form-item">
+        <label for="formMessage" class="form-label">
+          message
+        </label>
+        <textarea name="message" id="formMessage" class="form-input" v-model="formData.message"></textarea>
+      </div>
+      <button class="form-button" type="submit" :disabled="!isValid">
+        Send
+      </button>
+    </form> 
+  </div> 
+</template>
+
+<script>
+import moment from 'moment';
+import ValidationInput from '@/components/ValidationInput.vue';
+import { mapMutations, mapState } from 'vuex';
+
+export default {
+  name: 'FeedbackForm',
+  data: () => ({
+    inputNames: [
+      'name', 'surname', 'phone', 'email'
+    ],
+    formData: {
+      name: "", 
+      surname: "",
+      gender: "male",
+      phone: "",
+      birthday: "",
+      email: "",
+      agreement: false,
+      message: "",
+    },
+    formDataValidFlags: {
+      name: false, 
+      surname: false ,
+      phone: false,
+      email: false,
+      birthday: false,
+      agreement: false,
+    },
+    birthdayError: '',
+    birthErr: false,
+  }),
+  components: {
+    ValidationInput,
+  },
+  computed: {
+    ...mapState(['orderForms']),
+    isValid() {
+      return Object.values(this.formDataValidFlags).every(
+        (field) => field === true
+      );
+    },
+  },
+  methods: {
+    ...mapMutations({
+      setFeedBack: 'SET_FEED_BACK',
+    }),
+    checkBirthDate(event) {
+      if ( moment().diff(moment(event.target.value, 'YYYY-MM-DD').add(18,'years'), 'days' ) <= 1 || moment(event.target.value,'YYYY_MM_DD').isBefore('1923-01-01') || moment(event.target.value,'YYYY_MM_DD').isAfter('2023-01-01') ) {
+        this.birthdayError = 'Invalid date';
+        this.birthErr = true;
+      } else {
+        this.formDataValidFlags.birthday = moment().diff(moment(event.target.value, 'YYYY-MM-DD').add(18,'years'), 'days') > 1;
+        this.birthdayError = '';
+        this.birthErr = false;
+      }
+    },
+    getOrderNumber() {
+      if (!this.orderForms.length) 
+      return 1
+      let maxCount = 0;
+        for(let order of this.orderForms) {
+          maxCount = order.order_id > maxCount ? order.order_id : maxCount
+        }
+      return maxCount + 1;
+    },
+    submit() {
+      this.setFeedBack({
+        ...this.formData,
+        order_id: this.getOrderNumber()
+      });
+      this.formData.name="";
+      this.formData.surname="";
+      this.formData.gender="male";
+      this.formData.phone="";
+      this.formData.birthday="";
+      this.formData.email="";
+      this.formData.agreement="";
+      this.formData.message="";
```

Comment:
Object.keys(this.formData).forEach(field => { 
 this.formData[field] = field === 'gender' ? 'male' :  '';
} );